client/asset/btc/btc.go

```diff
@@ -5792,3 +5794,74 @@ func deserializeMsgTx(r io.Reader) (*wire.MsgTx, error) {
 func msgTxFromBytes(txB []byte) (*wire.MsgTx, error) {
 	return deserializeMsgTx(bytes.NewReader(txB))
 }
+
+func (btc *baseWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption, feeSuggestion uint64) (*asset.ConfirmRedemptionStatus, error) {
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+
+	_, confs, err := btc.rawWalletTx(txHash)
+	// redemption transaction found, return its confirms.
+	if err == nil {
+		return &asset.ConfirmRedemptionStatus{
+			Confs:  uint64(confs),
+			Req:    requiredRedeemConfirms,
+			CoinID: coinID,
+		}, nil
+	}
+
+	if !errors.Is(err, WalletTransactionNotFound) {
+		return nil, fmt.Errorf("problem searching for redemption transaction %s: %w", txHash, err)
+	}
+
+	// Redemption transaction is missing from the point of view of our node!
+	// Unlikely, but possible it was redeemed by another transaction. We
+	// assume a contract past its locktime cannot make it here, so it must
+	// not be refunded. Check if the contract is still an unspent output.
+
+	pkScript, err := btc.scriptHashScript(redemption.Spends.Contract)
+	if err != nil {
+		return nil, fmt.Errorf("error creating contract script: %w", err)
+	}
+
+	swapHash, vout, err := decodeCoinID(redemption.Spends.Coin.ID())
+	if err != nil {
+		return nil, err
+	}
+
+	utxo, _, err := btc.node.getTxOut(swapHash, vout, pkScript, time.Now().Add(-ContractSearchLimit))
+	if err != nil {
+		return nil, fmt.Errorf("error finding unspent contract %s: %w", redemption.Spends.Coin.ID(), err)
+	}
+	if utxo == nil {
+		// TODO: Spent, but by who. Find the spending tx.
+		btc.log.Warnf("Contract coin %v spent by someone but not sure who.", redemption.Spends.Coin.ID())
```

Comment:
Let's use `swapHash, vout` in this log and the two `error`s that use it.

Reply:
Adding them to the log and two error returns following obtaining it.